src/parser/jobs/drg/modules/Drift.tsx

```diff
@@ -0,0 +1,158 @@
+import {t} from '@lingui/macro'
+import {Trans} from '@lingui/react'
+import {Message, Table, Button, Header, Icon} from 'semantic-ui-react'
+import {ActionLink} from 'components/ui/DbLink'
+import ACTIONS from 'data/ACTIONS'
+import {CastEvent} from 'fflogs'
+import Downtime from 'parser/core/modules/Downtime'
+import {getDataBy} from 'data'
+import Module, {dependency} from 'parser/core/Module'
+import React, {Fragment} from 'react'
+import {Timeline} from 'parser/core/modules/Timeline'
+
+// Buffer (ms) to forgive insignificant drift, we really only care about GCD drift here
+// and not log inconsistencies / sks issues / misguided weaving
+const DRIFT_BUFFER = 1200
+
+const DRIFT_ABILITIES = [
+	ACTIONS.HIGH_JUMP.id,
+	ACTIONS.GEIRSKOGUL.id,
+]
+
+const COOLDOWN_MS = {
+	[ACTIONS.HIGH_JUMP.id]: ACTIONS.HIGH_JUMP.cooldown * 1000,
+	[ACTIONS.GEIRSKOGUL.id]: ACTIONS.GEIRSKOGUL.cooldown * 1000,
+}
+
+class DriftWindow {
+	actionId: number
+	start: number
+	end: number = 0
+	drift: number = 0
+	abilityRotation: CastEvent[] = []
+
+	constructor(actionId: number, start: number) {
+		this.actionId = actionId
+		this.start = start
+	}
+
+	public addAbility(event: CastEvent) {
+		const action = getDataBy(ACTIONS, 'id', event.ability.guid)
+		if (action) {
+			this.abilityRotation.push(event)
+		}
+	}
+
+	public getLastActionId(): number {
+		return this.abilityRotation.slice(-1)[0].ability.guid
+	}
+}
+
+export default class Drift extends Module {
+	static handle = 'drift'
+	static title = t('drg.drift.title')`Ability Drift`
+
+	@dependency private downtime!: Downtime
+	@dependency private timeline!: Timeline
+
+	private driftedWindows: DriftWindow[] = []
+
+	private currentWindows = {
+		[ACTIONS.HIGH_JUMP.id]: new DriftWindow(ACTIONS.HIGH_JUMP.id, this.parser.fight.start_time),
+		[ACTIONS.GEIRSKOGUL.id]: new DriftWindow(ACTIONS.GEIRSKOGUL.id, this.parser.fight.start_time),
+	}
+
+	protected init() {
+		this.addEventHook('cast', {by: 'player', abilityId: DRIFT_ABILITIES}, this.onDriftableCast)
+		this.addEventHook('cast', {by: 'player'}, this.onCast)
+	}
+
+	private onDriftableCast(event: CastEvent) {
+		let actionId: number
+		actionId = event.ability.guid
+
+		const window = this.currentWindows[actionId]
+		window.end = event.timestamp
+		const downtime = this.downtime.getDowntime(window.start, window.end)
+		const cd = COOLDOWN_MS[actionId]
+		window.drift = Math.max(0, window.end - window.start - cd - downtime)
```

Comment:
I'm a little concerned that this is going to just blanket forgive downtime that occurs during the cooldown, even if the skill never came off of cooldown during downtime (e.g. skill used at 1:00, downtime from 1:10-1:20, skill used again at 1:40 would wind up calculating this as 0s of drift, even though the skill became available at 1:30 and could have been used then).

Similarly, I'm not sure I understand what line 81 is doing - from the comment, I would have expected the compare to be downtime > cooldown.

Reply:
I'll admit I just kinda copy pasted the math from the MCH module, so I'll have to review it and maybe check it again. Do you have any suggestion for the first thing you pointed out? How to resolve/avoid that issue?

Reply:
Line 81 checks that the drift > buffer (drift is significant enough to report) and downtime < cd (downtime isn't longer than the CD itself, in which case when the downtime ends we'll likely have lots of "reopening" to do and we don't want to count this as drifting).

Reply:
As for line 76, I agree the calculation might be too lenient. If you have an idea for a more accurate implementation I would love to port it to the MCH module (and/or move it to core).

Reply:
In regards to:

> I haven't run this in local yet, so I'm basing this comment on screenshots posted in Discord, but I wonder if the presentation would be better if the data table had every usage (whether it drifted or not) and then showed 0s drift on the "on time" uses, more similar to what the SMN dot tables are displaying. It was a little jarring to me to see the screenshots with 3+ minute gaps between High Jump entries but only a few seconds of drift on the second use. 

I followed the suggestion and changed the table a bit, so it will have all the abilities showing, with a highlight per substantial drift. 
![image](https://user-images.githubusercontent.com/35241556/83822442-07da9d80-a69f-11ea-9331-49f0d5c41bee.png) In commit d86ae19.